docs/madr/decisions/056-meshservice-proxies-stats.md

```diff
@@ -0,0 +1,80 @@
+# MeshService proxies stats
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/10915
+
+## Context and Problem Statement
+
+[ServiceInsight](https://github.com/kumahq/kuma/blob/master/api/mesh/v1alpha1/service_insight.proto) has detailed information about the Service.
+We need an equivalent of it in MeshService.
+
+ServiceInsight includes the Status field
+* Online - all dataplane proxies are connected to CP and have healthy inbounds
+* Offline - all dataplane proxies are not connected or have no healthy inbounds
+* Partially degraded - not all are "online" and not all are "offline"
+
+This mixing of whether the DPP is connected to CP and whether is healthy is confusing.
+
+When aggregating multiple `MeshService` into `MeshMultiZoneService` we need to know to which zones we can send the traffic to.
+A zone can have `MeshService` but have no healthy endpoints, in this case we should try other zone instead.
+
+## Considered Options
+
+* Count in status
+* State in spec
+
+## Decision Outcome
+
+Chosen option: "Count in status" and "state in spec".
+
+## Pros and Cons of the Options
+
+### Count in status and state in spec
+
+```
+kind: MeshService
+status:
+  dataplaneProxies:
+    connected: 10
+    healthy: 8
+    total: 10
+```
+
+We now separate connected proxies to the zone control plane from proxies that have healthy inbounds.
+This information is computed by the source zone and synced to global for visibility.
+
+Fields:
+* `dataplaneProxies.connected` - number of DPPs connected to the Zone CP
+* `dataplaneProxies.healthy` - number of DPPs connected with all healthy inbounds
```

Comment:
Is it that the dpp has all healthy inbounds or inbounds for this service healthy?

Reply:
It's all inbounds selected by this service. At least for now.